### ISSUE TITLE
Datastore V2+ not raising networkStatus event and not syncronizing data when offline on iOS

### ISSUE THREAD

**Issue by filipesbragio** [aws-amplify/amplify-flutter#5217]
### Description

We've upgraded recently from Amplify V1 to V2
```
  amplify_api: ^1.8.0
  amplify_auth_cognito: ^1.8.0
  amplify_authenticator: ^1.0.0
  amplify_datastore: ^1.8.0
  amplify_flutter: ^1.8.0
```

to 

```
  amplify_api: ^2.3.0
  amplify_auth_cognito: ^2.3.0
  amplify_authenticator: ^2.1.1
  amplify_datastore: ^2.3.0
  amplify_flutter: ^2.3.0
```

Ever since this upgrade we haven't been receiving the networkStatus event when listening to datastore changes. Other events seem to be raised normally. Besides this, any changes we make while offline seem to be ignored and never updated on dynamoDB. Any changes made after becoming online again sync normally. I haven't created different issues for these as I believe they're related. I haven't been able to reproduce this on android, only on iOS.

We've also tried the 2.1.0 version and it seemed to have the same issue.

### Categories

- [ ] Analytics
- [ ] API (REST)
- [ ] API (GraphQL)
- [ ] Auth
- [ ] Authenticator
- [X] DataStore
- [ ] Notifications (Push)
- [ ] Storage

### Steps to Reproduce

on iOS, listen to data store events and turn on/off connection. No networkStatus events will be raised. 
Similarly, become offline and make changes to the model and then become online. The offline changes will not be synchronized. Subsequent changes while online will be synchronized normally.

```
StreamSubscription<DataStoreHubEvent> _createStreamSubscription() {
    return Amplify.Hub.listen(HubChannel.DataStore, (DataStoreHubEvent event) async {
      print(event.eventName);
    });
  }
```

### Platforms

- [X] iOS
- [ ] Android
- [ ] Web
- [ ] macOS
- [ ] Windows
- [ ] Linux

### Flutter Version

3.22.3

### Amplify Flutter Version

2.3.0

### Deployment Method

Amplify CLI

**Comment by Equartey** [comment]
Hi @filipesbragio, thank you for the detailed write up. We will investigate the root cause and give you updates as we have them.

**Comment by Equartey** [comment]
Hi @filipesbragio, quick update for you. 

I have opened https://github.com/aws-amplify/amplify-flutter/pull/5218 to resolve this issue, once it gets merged and released I'll notify you here. Thanks again for submitting this.

**Comment by filipesbragio** [comment]
Awesome! Thanks for the update.

**Comment by NikaHsn** [comment]
@filipesbragio the fix has been merged and will be released in the next version.

**Comment by NikaHsn** [comment]
I'm going to close this issue as the fix has been released in [2.4.0](https://github.com/aws-amplify/amplify-flutter/releases/tag/v2.4.0) version. If you see otherwise or have additional questions, please open a new issue.
